src/_pytest/_code/source.py

```diff
@@ -298,18 +307,16 @@ def getfslineno(obj) -> Tuple[Optional[Union["Literal['']", py.path.local]], int
         except TypeError:
             return "", -1
 
-        fspath = fn and py.path.local(fn) or None
+        fspath = fn and py.path.local(fn) or ""
```

Comment:
This cannot really happen (becoming `""`, only if `co_filename` is empty on a code object, but that gets handled by `Code` then, which I've fixed to keep it (instead of turning it into cwd).
But I guess the `or ""` does not hurt - at least I'm confident that it never returned `None` actually.